Modules/TRD/src/TrackletCountCheck.cxx

```diff
@@ -0,0 +1,189 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   TrackletCountCheck.cxx
+/// \author Deependra Sharma @IITB
+///
+
+#include "TRD/TrackletCountCheck.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "Common/Utils.h"
+// ROOT
+#include <TH1.h>
+#include <TPaveText.h>
+
+#include <DataFormatsQualityControl/FlagReasons.h>
+
+using namespace std;
+using namespace o2::quality_control;
+using namespace o2::quality_control_modules::common;
+
+namespace o2::quality_control_modules::trd
+{
+
+void TrackletCountCheck::configure()
+{
+  ILOG(Debug, Devel) << "initializing TrackletCountCheck" << ENDM;
+
+  mThresholdMeanHighPerTimeFrame = getFromConfig<float>(mCustomParameters, "UpperthresholdPerTimeFrame", 520.f);
+  ILOG(Debug, Support) << "using Upperthreshold Per Timeframe= " << mThresholdMeanHighPerTimeFrame << ENDM;
+
+  mThresholdMeanLowPerTimeFrame = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTimeFrame", 600.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Timeframe= " << mThresholdMeanLowPerTimeFrame << ENDM;
+
+  mThresholdMeanHighPerTrigger = getFromConfig<float>(mCustomParameters, "UpperthresholdPerTrigger", 520.f);
+  ILOG(Debug, Support) << "using Upperthreshold Per Trigger= " << mThresholdMeanHighPerTrigger << ENDM;
+
+  mThresholdMeanLowPerTrigger = getFromConfig<float>(mCustomParameters, "LowerthresholdPerTrigger", 500.f);
+  ILOG(Debug, Support) << "using Lowerthreshold Per Trigger= " << mThresholdMeanLowPerTrigger << ENDM;
+
+  mStatThresholdPerTrigger = getFromConfig<int>(mCustomParameters, "StatThresholdPerTrigger", 1000);
+  ILOG(Debug, Support) << "using StatThreshold Per Trigger= " << mStatThresholdPerTrigger << ENDM;
+}
+
+Quality TrackletCountCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  mResultPertrigger = Quality::Null;
+  mResultPerTimeFrame = Quality::Null;
+
+  for (auto& [moName, mo] : *moMap) {
+    ILOG(Debug, Support) << "moName =" << moName << ENDM;
+    (void)moName;
+    if (mo->getName() == "trackletsperevent") {
+      auto* h = dynamic_cast<TH1F*>(mo->getObject());
+      if (h == nullptr) {
+        // ILOG(Debug, Support) << "Requested Histogram type does not match with the Histogram in source" << ENDM;
+        continue;
+      }
+      // warning about statistics available
+      msg1 = std::make_shared<TPaveText>(0.3, 0.7, 0.7, 0.9, "NDC");
```

Comment:
```suggestion
if (!msg1) {
      msg1 = std::make_shared<TPaveText>(0.3, 0.7, 0.7, 0.9, "NDC");
      msg1->SetTextSize(10);
}
```

Reply:
In this way, it is inserting each time a new line into the Message pad. To use your suggestion I have to use ```msg1.reset()``` before allocating memory so that the text box does not overflow. Am I correct now?